### PR TITLE
Fixes for most bb_task permissions

### PR DESCRIPTION
**Pull request by rollick**

### CODE REVIEW

File: bluebottle/bb_tasks/permissions.py [hunk 1]
```diff
@@ -1,95 +1,75 @@
...
 
-    def has_object_permission(self, request, view, obj):
-        # Read permissions are allowed to any request, so we'll always allow GET, HEAD or OPTIONS requests.
-        if request.method in permissions.SAFE_METHODS:
+    def has_object_method_permission(self, method, user, view, obj):
+        if method in permissions.SAFE_METHODS:
             return True
 
-        if isinstance(obj, TaskMember) and obj.task.author == request.user:
+	# FIXME: when this permission is used with the update task member
```
Comment (rollick) [inline bluebottle/bb_tasks/permissions.py]: Not sure why the obj here is a Task when used on a TaskMember view

File: bluebottle/bb_tasks/permissions.py [hunk 2]
```diff
@@ -1,95 +1,75 @@
...
-
-        # Test for project model object-level permissions.
-        return isinstance(obj,
-                          TaskMember) and obj.member == request.user
-
-
-class IsMemberOrAuthorOrReadOnly(permissions.BasePermission):
 
+class MemberOrOwnerOrReadOnlyPermission(BasePermission):
+    # TODO: Move this check to the serialiser
```
Comment (rollick) [inline bluebottle/bb_tasks/permissions.py]: Still need to move this to the serializer

File: bluebottle/bb_tasks/permissions.py (hunk 1 above)
Comment (gannetson) [inline bluebottle/bb_tasks/permissions.py]: this is done in `BasePermission` where `obj` is replaced by `obj.parent`

**Comment by gannetson** [comment]
Merged into https://github.com/onepercentclub/bluebottle/pull/2838

**Pull request closed without merging**